src/app/shell/personal-cabinet/shared-cabinet/messages/messages.component.ts

```diff
@@ -0,0 +1,224 @@
+      new PushNavPath({
+        name: NavBarName.Messages,
+        isActive: false,
+        disable: true
+      })
+    );
+  }
+
+  getProviderWorkshops() {
+    this.store.dispatch(new GetWorkshopListByProviderId(this.providerId));
+  }
+
+  getChats() {
+    this.store.dispatch(new GetUserChatRooms(this.userRole));
+  }
+
+  setListeners() {
+    this.chatRooms.entities = this.mockChatRoom;
+    this.chatRooms.totalAmount = this.mockChatRoom.length;
+
+    this.chatRooms$
+      .pipe(
+        filter((chatRooms: SearchResponse<ChatRoom[]>) => !!chatRooms),
+        takeUntil(this.destroy$)
+      )
+      .subscribe((chatRooms: SearchResponse<ChatRoom[]>) => {
+        this.chatRooms = chatRooms;
+      });
+
+    this.filterFormControl.valueChanges
+      .pipe(
+        takeUntil(this.destroy$),
+        debounceTime(200),
+        distinctUntilChanged(),
+        filter((val: string) => !!val)
+      )
+      .subscribe((val: string) => {
+        //TODO: Implement search logic
+      });
+  }
+
+  onBlock(parentId: string): void {
+    const dialogRef = this.matDialog.open(ReasonModalWindowComponent, {
+      data: { type: ModalConfirmationType.blockParent }
+    });
+    dialogRef.afterClosed().subscribe((result: string) => {
+      if (result) {
+        const providerId = this.store.selectSnapshot<Provider>(
+          RegistrationState.provider
+        ).id;
+        const blockedParent = new BlockedParent(parentId, providerId, result);
+        this.store.dispatch(
+          new BlockParent(blockedParent, EntityType[this.subRole])
+        );
+      }
+    });
+  }
+
+  onUnBlock(parentId: string): void {
+    const dialogRef = this.matDialog.open(ConfirmationModalWindowComponent, {
+      width: Constants.MODAL_SMALL,
+      data: {
+        type: ModalConfirmationType.unBlockParent
+      }
+    });
+    dialogRef.afterClosed().subscribe((result: string) => {
+      if (result) {
+        const providerId = this.store.selectSnapshot<Provider>(
+          RegistrationState.provider
+        ).id;
+        const blockedParent = new BlockedParent(parentId, providerId);
+        this.store.dispatch(
+          new UnBlockParent(blockedParent, EntityType[this.subRole])
+        );
+      }
+    });
+  }
+
+  onEntitiesSelect(IDs: string[]): void {
+    //Need to be implemented when requests with parameters are made
+  }
+
+  //Delete after connecting to server
+  mockChatRoom: ChatRoom[] = [
+    {
+      id: 'mockId',
+      parentId: 'mockParentId',
+      workshopId: 'mockWorkShopId',
+      parent: {
+        id: 'mockParentId',
+        userId: 'mockUserId',
+        firstName: 'mockFName',
+        lastName: 'mockLName',
+        email: 'mockEmail',
+        phoneNumber: 'mockNumber',
+        middleName: 'mockMName'
+      },
+      workshop: {
+        id: 'mockWorkShopId',
+        providerTitle: 'mockProviderTitle',
+        title: 'mockTitle',
+        providerId: 'mockProviderId'
+      } as Workshop,
```

Comment:
mock data should be stored in assets, create a folder "mocks" 
![image](https://user-images.githubusercontent.com/27493959/197973380-4a54008c-437f-43ed-b64f-12a5534e1e02.png)
mock data in the final code should never be present, it always should be stored separately from the real working functionality